pkg/engine/executor/dataobjscan.go

```diff
@@ -203,36 +205,27 @@ func (s *dataobjScan) read() (arrow.Record, error) {
 		}
 	)
 
-	g, ctx := errgroup.WithContext(s.ctx)
-	g.SetLimit(max(runtime.GOMAXPROCS(0)/2, 1))
-
-	var gotData atomic.Bool
+	var gotData bool
 
-	for _, reader := range s.readers {
-		g.Go(func() error {
-			for {
-				buf := make([]logs.Record, 1024) // do not re-use buffer
-				n, err := reader.Read(ctx, buf)
-				if n == 0 && errors.Is(err, io.EOF) {
-					return nil
-				} else if err != nil && !errors.Is(err, io.EOF) {
-					return err
-				}
+	for {
+		buf := make([]logs.Record, 1024) // do not re-use buffer
+		n, err := s.reader.Read(context.Background(), buf)
+		if n == 0 && errors.Is(err, io.EOF) {
+			break
+		} else if err != nil && !errors.Is(err, io.EOF) {
+			return nil, err
+		}
 
-				gotData.Store(true)
+		gotData = true
 
-				heapMut.Lock()
-				for _, rec := range buf[:n] {
-					heap.Push(rec)
-				}
-				heapMut.Unlock()
-			}
-		})
+		heapMut.Lock()
```

Comment:
do we need this lock with only a single reader?

Reply:
nope, thanks for catching!